src/chunk.rs

```diff
@@ -1,12 +1,16 @@
-use crate::storage::{PlainStorage, Storage};
-use crate::{Error, ErrorKind, Tag, Version};
+use crate::{Error, ErrorKind, Tag};
 use byteorder::{BigEndian, ByteOrder, LittleEndian};
 use std::convert::TryFrom;
 use std::fmt;
-use std::fs;
-use std::io::prelude::*;
-use std::io::{BufReader, Seek, SeekFrom};
+use std::io::SeekFrom;
 use std::{convert::TryInto, io};
+#[cfg(feature = "encode")]
+use {
```

Comment:
Lots of changes like this, where various imports aren't used in the decode pathways.  It may make more sense to import some of these in the encode functions that actually use them rather than the ugly `cfg`-guarded import.